.github/workflows/ci.yml

```diff
@@ -48,6 +48,7 @@ jobs:
         uses: actions-rs/cargo@v1
         with:
           command: clippy
+          args: --all-targets --all-features -- -Dwarnings
```

Comment:
I don't know about this one. if you want to prevent something then we can also use the [`deny` attribute](https://doc.rust-lang.org/reference/attributes/diagnostics.html#lint-check-attributes) in the `lib.rs` root module to disallow specific lints. What might happen with this one is that your code stops building because clippy has introduced a new warning. We may be ok wit h that though.

Reply:
lets make it break in CI, but let's also make sure we don't get blocked once we actually need to get stuff merged.
there is a github-action [`on:` setting that schedules a regular build](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#schedule). we could build this thing at least once a week and that way find out that a rustc/clippy update added a new warning.

```yaml
on:
  schedule:
    - cron: "0 5 * * 4"
```

this should run our CI every thursday morning